libs/vm-modules/src/math/tensor/tensor_estimator.cpp

```diff
@@ -164,6 +179,36 @@ ChargeAmount TensorEstimator::Sum()
          COMPUTE_CHARGE_COST;
 }
 
+ChargeAmount TensorEstimator::ArgMax(SizeType const & /*indices*/)
+{
+  SizeType padded_size = fetch::math::Tensor<DataType>::PaddedSizeFromShape(tensor_.shape());
+  SizeType size        = fetch::math::Tensor<DataType>::SizeFromShape(tensor_.shape());
+
+  return static_cast<ChargeAmount>(DEFAULT_PADDED_SIZE_COEF * padded_size +
+                                   DEFAULT_SIZE_COEF * size + DEFAULT_CONST_COEF) *
+         COMPUTE_CHARGE_COST;
+}
+
+ChargeAmount TensorEstimator::ArgMaxNoIndices()
+{
+  SizeType padded_size = fetch::math::Tensor<DataType>::PaddedSizeFromShape(tensor_.shape());
+  SizeType size        = fetch::math::Tensor<DataType>::SizeFromShape(tensor_.shape());
+
+  return static_cast<ChargeAmount>(DEFAULT_PADDED_SIZE_COEF * padded_size +
+                                   DEFAULT_SIZE_COEF * size + DEFAULT_CONST_COEF) *
+         COMPUTE_CHARGE_COST;
+}
+
+ChargeAmount TensorEstimator::Dot(vm::Ptr<VMTensor> const & /*other*/)
+{
+  SizeType padded_size = fetch::math::Tensor<DataType>::PaddedSizeFromShape(tensor_.shape());
+  SizeType size        = fetch::math::Tensor<DataType>::SizeFromShape(tensor_.shape());
+
+  return static_cast<ChargeAmount>(DEFAULT_PADDED_SIZE_COEF * padded_size +
+                                   DEFAULT_SIZE_COEF * size + DEFAULT_CONST_COEF) *
+         COMPUTE_CHARGE_COST;
+}
```

Comment:
these three methods above and .Copy() share identical code, so it would be great to just have 1 and and call it.

Reply:
it probably won't stay that way